### PR TITLE
Selection page

### PR DESCRIPTION
**Pull request by ConorMurphy21**
- add an active selection and passive selection page
 - add backend to accept selection from correct selector
 - add backend to handle state machine of rotating selections
 - add unit tests to test backend

### CODE REVIEW

File: client/src/components/gameShared/ResponseList.vue [hunk 1]
```diff
@@ -1,19 +1,46 @@
...
+           :class="{'list-group-item-action': selectable,
+                    'active': selected === index}"
+           @click="select(index, response)">
+        {{ response }}
+      </div>
+    </div>
+  </div>
 </template>
 <script>
 import {createNamespacedHelpers} from "vuex";
```
Comment (SatireSage) [inline client/src/components/gameShared/ResponseList.vue:15]: move import to the top

File: client/src/components/selection/ActiveSelection.vue [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+<template>
+  <div class="w-100 h-100 d-flex flex-column justify-content-between align-items-center p-3">
+    <h1>{{ prompt }}</h1>
+    <selection-type/>
+    <response-list :selectable="true" v-model="response"/>
+  </div>
+</template>
+
+<script>
+import {createNamespacedHelpers} from "vuex";
```
Comment (SatireSage) [inline client/src/components/selection/ActiveSelection.vue:15]: move import to the top, if possible

**Review by SatireSage — APPROVED** [review]
Looks good, the state machine eliminates hard coding. Maybe add more comments if possible!

File: client/src/components/selection/ActiveSelection.vue (hunk 2 above)
Comment (ConorMurphy21) [reply client/src/components/selection/ActiveSelection.vue:15]: It is at the top of the script section

File: client/src/components/gameShared/ResponseList.vue (hunk 1 above)
Comment (ConorMurphy21) [reply client/src/components/gameShared/ResponseList.vue:15]: Same as above

**Pull request merged**